### PR TITLE
Allow user to specify hoverinfo

### PR DESCRIPTION
**Pull request by nickmelnikov82**
plotly/dash-bio#519

### CODE REVIEW

File: app/scripts/fornac.js [hunk 1]
```diff
@@ -1654,6 +1657,19 @@ export function FornaContainer(element, passedOptions) {
         self.updateStyle();
     };
 
+    self.getGetTitleText = function (d) {
```
Comment (HammadTheOne) [inline app/scripts/fornac.js]: Can we change this to just `getTitleText` instead?

Comment (nickmelnikov82) [reply app/scripts/fornac.js]: Yes, sure. Sorry, that's my bad.

**Review by HammadTheOne — COMMENTED** [review]
This looks pretty good, just a couple of points before merging. 

I think calling the prop `hoverPattern` instead of `titlePattern` might be a more accurate description, and would make a little more sense in this case. 

It seems like the default `structName` key is not set; if I leave the prop unassigned in my component, the hovertext is `empty - 3` rather than the  base name. Setting the prop to `'${name}:${num}'` instead does give us the correct hovertext (eg. `A - 3`). 

image

And lastly, I think we should explicitly state in the prop description which keys are accepted for this pattern. If it's the following node properties, we should include a list of them in the prop description and possibly output a warning if unsupported keys are passed to the pattern.

```
self.nodes = [{ 'name': 'P',
	        'num': 1,
	        'radius': 3 * Math.sqrt(size),
	        'rna': self,
	        'nodeType': 'protein',
	        'structName': structName,
	        'elemType': 'p',
	        'size': size,
	        'uid': generateUUID() }];
```

**Comment by nickmelnikov82** [comment]
> It seems like the default structName key is not set; if I leave the prop unassigned in my component, the hovertext is empty - 3 rather than the base name. Setting the prop to '${name}:${num}' instead does give us the correct hovertext (eg. A - 3).

By default, we use the 'empty' structName property. So if custom structName isn't passed, the `empty:3` hover text is correct.

**Comment by nickmelnikov82** [comment]
@HammadTheOne I've renamed props and added console warning if unsupported keys are passed to the pattern.

**Comment by HammadTheOne** [comment]
> By default, we use the 'empty' structName property. So if custom structName isn't passed, the empty:3 hover text is correct.

I see, so that way if we have multiple molecule sequences, it will show their structure name's by default on hover. That makes sense.

In that case, can we maybe use a ternary operator to set the default hoverTest, checking to see if `structName` is passed or not, and if it is not, we can default it to maybe `'${nodeType}:${num}'` instead?

**Comment by nickmelnikov82** [comment]
> In that case, can we maybe use a ternary operator to set the default hoverTest, checking to see if structName is passed or not, and if it is not, we can default it to maybe '${nodeType}:${num}' instead?

Yes, we can. In this case, we should remove default props from `fornactontainer` component in dash_bio and implement this functionality in this repo.

**Comment by nickmelnikov82** [comment]
@HammadTheOne 
image

Can you please check my function for getting hoverPattern? Is my understanding correct?

**Comment by HammadTheOne** [comment]
@nickmelnikov82 That looks good to me, the only change I would make is changing `name` in the `defaultPattern` to `nodeType` instead.

**Review by HammadTheOne — APPROVED** [review]
:woman_dancing: 

Tested it out, and the `hoverPattern` default works like we described. Fantastic!

**Comment by nickmelnikov82** [comment]
@HammadTheOne so, can we merge this PR?

**Pull request merged**